neps/nep-0393.md

```diff
@@ -0,0 +1,480 @@
+        &self,
+        ctr: AccountId,
+        account: AccountId,
+        class: Option<ClassId>,
+    ) -> u64;
+
+    /// Query sbt tokens issued by a given contract.
+    /// If `from_index` is not specified, then `from_index` should be assumed
+    /// to be the first valid token id.
+    fn sbt_tokens(
+        &self,
+        ctr: AccountId,
+        from_index: Option<u64>,
+        limit: Option<u32>,
+    ) -> Vec<TokenId>;
+
+    /// Query SBT tokens by owner
+    /// If `from_class` is not specified, then `from_class` should be assumed to be the first
+    /// valid class id.
+    /// Returns list of pairs: `(Contract address, list of token IDs)`.
+    fn sbt_tokens_by_owner(
+        &self,
+        account: AccountId,
+        ctr: Option<AccountId>,
+        from_class: Option<u64>,
+        limit: Option<u32>,
+    ) -> Vec<(AccountId, Vec<TokenId>)>;
+
+    /// checks if an `account` was banned by the registry.
+    fn is_banned(&self, account: AccountId) -> bool;
+
+
+    /*************
+     * Transactions
+     *************/
+
+    /// Creates a new, unique token and assigns it to the `receiver`.
+    /// `token_spec` is a vector of pairs: owner AccountId and TokenMetadata.
+    /// Each TokenMetadata must have non zero `class`.
+    /// Must be called by an SBT contract.
+    /// Must emit `Mint` event.
+    /// Must provide enough NEAR to cover registry storage cost.
+    // #[payable]
+    fn sbt_mint(&mut self, token_spec: Vec<(AccountId, TokenMetadata)>) -> Vec<TokenId>;
+
+    /// sbt_recover reassigns all tokens from the old owner to a new owner,
+    /// and registers `old_owner` to a burned addresses registry.
+    /// Must be called by an SBT contract.
+    /// Must emit `Recover` event.
+    /// Must provide enough NEAR to cover registry storage cost.
+    /// Requires attaching enough tokens to cover the storage growth.
+    // #[payable]
+    fn sbt_recover(&mut self, from: AccountId, to: AccountId);
+
+    /// sbt_renew will update the expire time of provided tokens.
+    /// `expires_at` is a unix timestamp (in seconds).
+    /// Must be called by an SBT contract.
+    /// Must emit `Renew` event.
+    fn sbt_renew(&mut self, tokens: Vec<TokenId>, expires_at: u64);
+
+    /// Revokes SBT, could potentially burn it or update the expire time.
+    /// Must be called by an SBT contract.
+    /// Must emit one of `Revoke` or `Burn` event.
+    /// Returns true if a token is a valid, active SBT. Otherwise returns false.
+    fn sbt_revoke(&mut self, token: u64) -> bool;
+
+    /// Transfers atomically all SBT tokens from one account to another account.
+    /// The caller must be an SBT holder and the `to` must not be a banned account.
+    /// Must emit `Revoke` event.
+    // #[payable]
+    fn sbt_soul_transfer(&mut self, to: AccountId) -> bool;
+}
+```
+
+SBT smart contracts can implement NFT query interface to make it compatible with NFT tools. Note, we use U64 type rather than U128.
```

Comment:
I don't think we should suggest anyone implementing a fallback "query interface" without specifying explicitly what we expect this to be used for and by whom.

If we're not 100% sure, I'd suggest removing this section all-together, especially since the community decided to not emit NFT specific events.